osu.Game/Screens/Edit/Compose/Components/Timeline/ZoomableScrollContainer.cs

```diff
@@ -136,11 +136,20 @@ private void setZoomTarget(float newZoom, float focusPoint)
         {
             zoomTarget = Math.Clamp(newZoom, MinZoom, MaxZoom);
             transformZoomTo(zoomTarget, focusPoint, ZoomDuration, ZoomEasing);
+
+            OnZoomChange();
         }
 
         private void transformZoomTo(float newZoom, float focusPoint, double duration = 0, Easing easing = Easing.None)
             => this.TransformTo(this.PopulateTransform(new TransformZoom(focusPoint, zoomedContent.DrawWidth, Current), newZoom, duration, easing));
 
+        /// <summary>
+        /// Invoked when the zoom target has changed.
+        /// </summary>
+        protected virtual void OnZoomChange()
+        {
+        }
```

Comment:
Not sure if it's just me but isn't this method name sort of misleading, requiring the xmldoc to correct it? It's called on zoom *target* change, not zoom change. And that's a pretty significant difference due to the zoom transforming.

It's even more of a mess what with `ZoomableScrollContainer` publically exposing `Zoom` and `CurrentZoom` but internally using `zoomTarget` and `currentZoom`...

Also these sorts of event handlers usually use the past tense in the name (i.e. `OnZoomTargetChanged()`).

Reply:
Was following with what `ZoomableScrollContainer` publicly exposes (`Zoom`), might even make sense rewriting the xmldoc to `Invoked when <see cref="Zoom"/> has changed` so that the entirety of "zoom target" is kept private to the implementation itself.

As for using past tense, I was thinking about `Drawable`'s input event handler names, but now that I look again, almost all event handlers do use past tense names...

Reply:
I would honestly prefer the public member to be called `TargetZoom` too to decrease ambiguity but that's probably out of scope here.